### ISSUE TITLE
PH Graph/Probe Reading Issues since 6.0

### ISSUE THREAD

**Issue by danmrossi** [reef-pi/reef-pi#1977]
Hi @ranjib 

Been liasing with "tdragon" with his Reef-Pi Home Assistant Add-on "https://github.com/tdragon/reef-pi-hass-custom/issues/20" and we have noticed that when i pull the readings from "http://192.168.1.202/api/phprobes/2/readings" - that ip address being the Reef-Pi, the ordering of the readouts is not ordered... 

As per tdragons comment in my issues thread in his github: "Look at the time - it is not ordered. The integration gets the latest line from the "current" list, but if it is not ordered it can be wrong. I'll try to sort all the data, but it seems to be a bug in ReefPi."

Are you able to shed any light on this?

Also since going from 5.3 to 6.0 the ph graph goes all funky for lack of a better word, showing the odd spike to 0 or some really crazy negative number. if i go back to 5.3 that stops.

btw i am using Robo-Tank Controller setup with robo-tank install script thats been working fine for last 8 months till now using a Raspberry Pi Zero 2W with Raspbian "buster". Should i be using "bulleseye" or that shouldn't make a difference?

Any ideas? Really wanting to use 6.0 as i want to utilise some ESP32 devices but the above is making it difficult for me lol

Many thanks

Daniel

**Comment by danmrossi** [comment]
Hi @ranjib 

This is the difference between Reef-Pi v5.3 and v6.0 is shown in the PH Graph Below.

![IMG_0032](https://user-images.githubusercontent.com/50608357/204191926-1976aa4e-32bc-4dfa-8371-c79c3e81036e.PNG)

**Comment by danmrossi** [comment]
[reef-pi log.txt](https://github.com/reef-pi/reef-pi/files/10109932/reef-pi.log.txt)

Attached is the log file from my reef-pi showing that the PH reading has been flawless.

I have tried the following to no avail, each time with a reinstall of raspbian buster and different sd cards as well:
- Installed Reef-Pi 5.2 with Robo-Tank Script
- Tried updating direct from 5.2 to 6.0 using backup copy of my reef-pi db - same effect as above.
- Tried setup with a default reef-pi db from RoboTank script and also setting up new one / controller from scratch without any preconfigured db to rule out any corruption - same effect as above
- Tried install of 6.0 direct and manual config - same effect as above
- Verified no power cables are running anywhere near the Reef-Pi Controller or the PH Probe and its Cable.
- No other hardware has changed in months on or near my aquarium or changed placement inside or outside or around the aquarium or room.
- Verified with [@robsworld78](https://www.reef2reef.com/members/142670/) my setup of config.txt etc and no parameters needed to be changed
- I checked daemon logs, journal logs etc (spent hours going through) and can see that each time reef-pi requests an update from the ph probe it gets a valid reading.
- via the API call to my Home Assistant instance on another Pi in the house, the requests/graph there come through perfect without any issue.

**Comment by ranjib** [comment]
Hi Daniel
Thank you so much for the details. If the values are not changed that mean
it’s not the controller software , but the graphing library that’s causing
the issue. Don’t reinstall anything as clearly those won’t solve any issue
, since it’s due to the graphing library upgrade. All major release
contains numerous updates and sometime bugs like this gets introduced in
the process. While i debug this (I think it’s some configuration in rechart
library that I have to tune ) , I think there’s not functional negative
impact from this , is that correct ?

**Comment by ranjib** [comment]
One more thing , if you hover your mouse cursor over the dips in the graph
, do they show any value ?

**Comment by danmrossi** [comment]
Hey Ranjib thats ok :) Some of the dips do show as 0.16 when i can get the cursor over the dip properly but not all show a reading.

No functional negative impact as of yet that i can see as at the moment im not having reef-pi control my co2 system just read ph so while im away i can see whats going on

**Comment by danmrossi** [comment]
fyi i did just create a post in reef to reef with the same questions / issues as above as didnt know if was easier / better to get yours or others attention on this here or there :P

**Comment by robsworld78** [comment]
Hey Ranjib, it never ends. 😊 😊 😊 

 

Hope all is well with you and your family and you have a great holiday season, that’s the plan here. 😊

**Comment by danmrossi** [comment]
Hey @ranjib can you pls confirm you received this update/comment? :)

I have also found that since now the Reef-Pi-Hass-Custom Addon now pulls it's PH info from http://192.168.1.202/api/phprobes/2/read as in v6.0 apparently the guy who makes Reef-Pi-Hass-Custom was not pulling information from there before because it wasn't working correctly in previous reef-pi builds or something: 

Also found that since v5.2 all the way to the latest 6.0 release your ph graphs aren't displaying the correct ph reading for me. i know this to be true as a 2nd separate/standalone ph meter i have running concurrently on my tank at the time of writing this displays the value of 7.38 yet your graph in reef-pi states the current PH is 7.60

So in other words, I get a completely different reading from the Home Assistant addon getting the ph reading from the API which matches my separate ph controller reading at the same time yet your readout and graph from within reef-pi has been off since 5.2 for me and is still off in 6.0

**Comment by danmrossi** [comment]
Hi @ranjib i hope you are well.

Just wondering when this will be looked at?

I also noticed this issue makes it a bit difficult to calibrate probes when the reading drops off when in the Reef-Pi GUI.

Also makes the graph useless and trying to use any automations against the ph reading pointless as well.

If i could make heads or tails of the code i'd fix it myself but alas i am at your mercy  :)

**Comment by Cogginshome** [comment]
Having same problem it's not just a graph I run co based on ph and it has a fit every time it swings down to 0.whatever 
But I'm on 5.3 also running robotank hardware

**Comment by danmrossi** [comment]
Hi @ranjib i was referring to this ticket when I mentioned you in the “ranjib updates” ticket

Should I just close the ranjib updates ticket?
